libpod/kube.go

```diff
@@ -119,6 +119,61 @@ func (p *Pod) getInfraContainer() (*Container, error) {
 	return p.runtime.GetContainer(infraID)
 }
 
+func GenerateForKubeDaemonSet(ctx context.Context, pod *YAMLPod, options entities.GenerateKubeOptions) (*YAMLDaemonSet, error) {
+	// Restart policy for DaemonSets can only be set to Always
+	if options.Type == define.K8sKindDaemonSet && !(pod.Spec.RestartPolicy == "" || pod.Spec.RestartPolicy == v1.RestartPolicyAlways) {
```

Comment:
Isn't is assumed that this method is called only for `options.Type == define.K8sKindDaemonSet`? Why would it be called otherwise?

Reply:
You're right, I copied over some of it from the Deployment implementation and it remained. Fixed.